p2p/host/basic/basic_host.go

```diff
@@ -23,6 +23,8 @@ var log = logging.Logger("basichost")
 
 var NegotiateTimeout = time.Second * 60
 
+type AddrsFactory func([]ma.Multiaddr) []ma.Multiaddr
+
 // Option is a type used to pass in options to the host.
```

Comment:
I counted and there's at least 4 different ways of passing in various options to host and swarm, including global variables :) -- it's in bad need of an options struct.